lib/Settings/Settings.js

```diff
@@ -61,7 +61,8 @@ class Settings extends Component {
     // links in a given section
     const navlinks = (section) => {
       const navItems = section.pages.map((p) => {
-        const link = `${props.match.path}/${p.route}`;
+        const search = props?.location?.search || '';
```

Comment:
`location` and `search` are always available.
```suggestion
        const search = props.location.search || '';
```